src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexWriter.cs

```diff
@@ -73,6 +72,9 @@ public void Commit(IState state)
                 if (e is Win32Exception win32Exception && win32Exception.IsOutOfMemory())
                     ThrowOutOfMemory(e);
 
+                if (e.InnerException is VoronUnrecoverableErrorException)
+                    throw e.InnerException;
```

Comment:
We are now losing the inner stack here.
Maybe throw that in a *good* exception?

Reply:
Do we need the inner stack? (this exception is wrapped)
https://github.com/ravendb/lucenenet/blob/3.0.3-netcore/src/Lucene.Net/Index/IndexWriter.cs#L4644

Reply:
We never want to lose the stack

Reply:
done